ddtrace/tracer/abandoned_spans_test.go

```diff
@@ -0,0 +1,219 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var warnPrefix = fmt.Sprintf("Datadog Tracer %v WARN: ", version.Tag)
+var spanStart = time.Date(2023, time.August, 18, 0, 0, 0, 0, time.UTC)
+
+// setTestTime() sets the current time, which will be used to calculate the
+// duration of abandoned spans.
+func setTestTime() {
```

Comment:
a nit here is instead of having a separate reset function you can just return that reset function as a func here so then any test that needs to set the test time can do so (and reset) by just `defer setTestTime()()`